src/Routes/Signatures/Details.js

```diff
@@ -70,6 +70,45 @@ const Details = () => {
                             {!isCodeEditorEnabled && <GridItem xl2={7} xl={3} md={6} xs={12}>
                                 {detailBlock(intl.formatMessage(messages.description), sigDetailsData?.metadata?.description)}
                             </GridItem>}
+                            <GridItem xl2={isCodeEditorEnabled && 6 || 1} xl={isCodeEditorEnabled && 6 || 3} md={6} xs={12}>
```

Comment:
I see that there are some repetitions. We can optimise them by creating a new component to render a single whole GridItem. It receives only message and renders the whole GridItem with dataNotAvailable message conditionally.

Reply:
I have reduced some code repetition but it isn't quite what you were referring to in your comment @mkholjuraev .  Please have a look at my new commit anyway and see what you think.  Thanks.